### PR TITLE
Add signature v2 support for S3 chunks client

### PR DESCRIPTION
**Pull request by simonswine**
**What this PR does**:

* This allows to configure the chunks S3 client to use V2 signatures for S3 authentication (is also used for the alertmanager and ruler store).

* This follows the interface of #3540 for the blocks client

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/chunk/grpc/grpc.pb.go [hunk 1]
```diff
@@ -7,17 +7,18 @@ import (
 	bytes "bytes"
 	context "context"
 	fmt "fmt"
+	io "io"
+	math "math"
+	math_bits "math/bits"
+	reflect "reflect"
+	strings "strings"
+
```
Comment (jtlisi) [inline pkg/chunk/grpc/grpc.pb.go:15]: **nit:** Not sure you meant to commit this. This looks like a autoformatter hitting a generated file.

**Review by jtlisi — APPROVED** [review]
LGTM

There is one nit that needs to be fixed before this can be merged.

File: pkg/chunk/grpc/grpc.pb.go (hunk 1 above)
Comment (simonswine) [reply pkg/chunk/grpc/grpc.pb.go]: Yes that was a too aggressive `go-imports` run

**Review by pstibrany — APPROVED** [review]
LGTM

**Review by pracucci — COMMENTED** [review]
Post-merge comment: good job! 👏

**Pull request merged**